### ISSUE TITLE
Documentation, how to send a dynamic update with/without TSIG

### ISSUE THREAD

**Issue by tkandal** [miekg/dns#1166]
Dear, Sirs!

Could you please update the documentation to include
comprehensive and complete examples of how send a dynamic
update with/without TSIG?
I have been trying with the API for several days, but I don't understand
how to do this.

This would be very much appreciated!

Thank you!

**Comment by miekg** [comment]
what have *you* tried, where did you get stuck, what code didn't work?

**Comment by tkandal** [comment]
Hello, Sir!

Since I was completely ignorant regarding dynamic updates of DNS, I was in deep water already in the beginning.  I read the documentation many times, but because there was not a complete example of how to add, update or delete DNS-records I got very confused.  A colleague of mine also insisted that I needed to use IXFR-call(s) to do these operations, which also confused me even more.

After a week I discovered that I could find examples in update_test.go, which were exactly what I needed.  Then everything fell into place.

Anyhow, I believe it would be a generally good idea to at least include one complete example of how to do dynamic update in the code-examples.  The idea is to get users of the library, who are not bleeding experts on DNS and the belonging protocol up to speed faster.  Many programmers, like myself, jump from project to project, and sadly we are not always able or have the time to examine and understand all the underlying details.

I hope you appreciate the idea.  Thank you very much!
